src/visual.ts

```diff
@@ -1215,40 +1233,760 @@ export class MekkoChart implements IVisual {
+                            control: {
+                                type: powerbi.visuals.FormattingComponent.FontControl,
+                                properties: {
+                                    fontFamily: {
+                                        descriptor: {
+                                            objectName: "legend",
+                                            propertyName: "fontFamily"
+                                        },
+                                        value: legendFontFamily
+                                    },
+                                    fontSize: {
+                                        descriptor: {
+                                            objectName: "legend",
+                                            propertyName: "fontSize"
+                                        },
+                                        value: legendFontSize
+                                    }
+                                }
+                            }
+                        },
+                    ]
+                }
+            ]
+        };
+        legendCard.groups.push(this.getSortLegendFormattingGroup());
+        return legendCard;
+    }
+
+    private getSortLegendFormattingGroup(): powerbi.visuals.FormattingGroup {
+        const legendSortSettings: MekkoLegendSortSettings = (<BaseColumnChart>this.layers[0]).getLegendSortSettings();
+        let sortLegendGroup: powerbi.visuals.FormattingGroup = {
+            displayName: "Sort Legend",
+            uid: "legendCard_sortLegend_group_uid",
+            slices: [
+                {
+                    uid: "legendCard_sortLegend_enable_uid",
+                    displayName: "Enabled",
+                    control: {
+                        type: powerbi.visuals.FormattingComponent.ToggleSwitch,
+                        properties: {
+                            descriptor: {
+                                objectName: "sortLegend",
+                                propertyName: "enabled"
+                            },
+                            value: legendSortSettings.enabled
+                        }
+                    }
+                },
+                {
+                    uid: "legendCard_sortLegend_direction_uid",
+                    displayName: "Direction",
+                    control: {
+                        type: powerbi.visuals.FormattingComponent.Dropdown,
+                        properties: {
+                            descriptor: {
+                                objectName: "sortLegend",
+                                propertyName: "direction"
+                            },
+                            value: legendSortSettings.direction
+                        }
+                    }
+                },
+                {
+                    uid: "legendCard_sortLegend_groupbyCategory_uid",
+                    displayName: "Group By Category",
```

Comment:
As all text lines are localized, you'll have to add displayNameKey to every property. If you create a new one, don't forget to add it to [https://github.com/microsoft/powerbi-visuals-mekkochart/blob/master/stringResources/en-US/resources.resjson](https://github.com/microsoft/powerbi-visuals-mekkochart/blob/master/stringResources/en-US/resources.resjson)